### PR TITLE
Document datadog extension

### PR DESCRIPTION
**Pull request by mackjmr**
### What does this PR do?

This PR adds a readme to the datadog extension component. For the zpages and pprof extensions, seems like the current logic is broken so need to wait before adding a description of contents.

### CODE REVIEW

File: comp/otelcol/extension/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+### Environment variables
+
+The flare will collect all environment variables, and these can be found in `otel/otel-flare/environment.json`.
+
+### Extension data
+
+The flare also adds data collected from extensions. These extensions are added automatically by the [converter component](../converter/README.md). The data collected is from extensions:
+- health_check: Found in `otel/otel-flare/health_check`. 
+
+Will contain a json of the latest health check, for example:
```
Comment (mx-psi) [inline comp/otelcol/extension/README.md:42]:
```suggestion
Will contain a JSON of the latest health check, for example:
```

**Review by mx-psi — APPROVED** [review]

**Review by hush-hush — APPROVED** [review]

File: comp/otelcol/extension/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# Datadog Extension
+
+The datadog extension component allows inclusion of otel-agent data in the datadog-agent [flare](https://docs.datadoghq.com/agent/troubleshooting/send_a_flare/?tab=agent). A flare can be triggered by the core agent process, or remote-config.
```
Comment (truthbk) [inline comp/otelcol/extension/README.md:3]: We should also add how the extension provides the relevant metadata for `otel-agent` configuration and inventory tracking in Fleet Automation. This metadata is periodically collected by the core-agent which then submits that data to the backend..

File: comp/otelcol/extension/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+*Agent Config:*
+```
+otel-agent:
+ enabled: true
+ flare_port: 7777
+```
+
+The port is the location in which the otel-agent will expose the data required to build the flare. The core agent then fetches the data from this port.
+
+## Data collected by flare
```
Comment (truthbk) [inline comp/otelcol/extension/README.md:25]: We may want to add a small section below this one about the additional metadata submitted for fleet automation, which includes stuff like the `otel-agent` version, etc.

**Review by truthbk — COMMENTED** [review]
Looks great, just a small note that we should also mention the role of the extension in the submission of the Fleet Automation metadata too.

File: comp/otelcol/extension/README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+### Configurations
+
+The flare will collect both the provided collector config and the enhanced config (enhanced via [converter](../converter/README.md)). 
+
+The provided collector configs can be found in `otel/otel-flare/customer.cfg` and the enhanced config can be found in `otel/otel-flare/customer.cfg`.
+
+### Environment variables
+
```
Comment (dineshg13) [inline comp/otelcol/extension/README.md:34]: Lets add the diagram from RFC as well

File: comp/otelcol/extension/README.md (hunk 2 above)
Comment (mackjmr) [reply comp/otelcol/extension/README.md]: Added, can you please take a look ?

File: comp/otelcol/extension/README.md (hunk 3 above)
Comment (mackjmr) [reply comp/otelcol/extension/README.md]: same as above 👍

File: comp/otelcol/extension/README.md (hunk 4 above)
Comment (mackjmr) [reply comp/otelcol/extension/README.md]: Added

**Review by truthbk — APPROVED** [review]
This looks great! :shipit:

**Pull request merged**